multitenantccl: leaked goroutine in `TestDataDriven/cleanup`

```
--- FAIL: TestDataDriven (5.52s)
    test_log_scope.go:161: test logs captured to: /Users/rc/src/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/50e33444d6d65b5a0bf1480b8c91c2a1/logTes
tDataDriven1245022510
    test_log_scope.go:79: use -show-logs to present logs inline
    --- FAIL: TestDataDriven/cleanup (5.50s)
        server_test.go:60: Leaked goroutine: goroutine 13 [select]:
            github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).gcDaemon(0x14000cbe8f0, {0x104f17750, 0x1400076cec0})
                github.com/cockroachdb/cockroach/pkg/util/log/file_log_gc.go:25 +0x70
            created by github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig
                github.com/cockroachdb/cockroach/pkg/util/log/flags.go:306 +0xdc8
            Leaked goroutine: goroutine 14 [select]:
            github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).gcDaemon(0x14000cbe9c0, {0x104f17750, 0x1400076cec0})
                github.com/cockroachdb/cockroach/pkg/util/log/file_log_gc.go:25 +0x70
            created by github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig
                github.com/cockroachdb/cockroach/pkg/util/log/flags.go:306 +0xdc8
            Leaked goroutine: goroutine 15 [select]:
            github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).gcDaemon(0x14000cbea90, {0x104f17750, 0x1400076cec0})
                github.com/cockroachdb/cockroach/pkg/util/log/file_log_gc.go:25 +0x70
            created by github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig
                github.com/cockroachdb/cockroach/pkg/util/log/flags.go:306 +0xdc8
            Leaked goroutine: goroutine 16 [select]:
            github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).gcDaemon(0x14000cbeb60, {0x104f17750, 0x1400076cec0})
                github.com/cockroachdb/cockroach/pkg/util/log/file_log_gc.go:25 +0x70
            created by github.com/cockroachdb/cockroach/pkg/util/log.ApplyConfig
                github.com/cockroachdb/cockroach/pkg/util/log/flags.go:306 +0xdc8
    server_test.go:61: -- test log scope end --
```

Problem is reproducible in a few seconds by stressing this test on `master`.

Jira issue: CRDB-22348

--- a/pkg/ccl/multitenantccl/tenantcostserver/server_test.go
+++ b/pkg/ccl/multitenantccl/tenantcostserver/server_test.go
@@ -41,10 +41,10 @@ import (
 
 func TestDataDriven(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
 
 	datadriven.Walk(t, testutils.TestDataPath(t), func(t *testing.T, path string) {
 		defer leaktest.AfterTest(t)()
+		defer log.Scope(t).Close(t)
 
 		var ts testState
 		ts.start(t)

